File: solutions/_617_merge_two_binary_trees/Solution.java

package _617_merge_two_binary_trees;

class Solution {
    public TreeNode mergeTrees(TreeNode a, TreeNode b) {
        if (a == null && b == null)
            return null;
        if (a == null)
            return b;
        if (b == null)
            return a;

        a.val += b.val;
        a.left = mergeTrees(a.left, b.left);
        a.right = mergeTrees(a.right, b.right);

        return a;
    }
}

/*
 * https://leetcode.com/problems/merge-two-binary-trees/
 */

File: solutions/_617_merge_two_binary_trees/solution.py
from typing import Optional


# Definition for a binary tree node.
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


class Solution:
    def mergeTrees(
        self, a: Optional[TreeNode], b: Optional[TreeNode]
    ) -> Optional[TreeNode]:
        if not a and not b:
            return None
        if not a:
            return b
        if not b:
            return a
        a.val += b.val
        a.left = self.mergeTrees(a.left, b.left)
        a.right = self.mergeTrees(a.right, b.right)
        return a


# https://leetcode.com/problems/merge-two-binary-trees/
